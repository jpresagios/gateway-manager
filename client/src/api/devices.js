import axios from "axios";
import env from "../configs/env-config";

function deleteDevice(idDevice) {
  return axios
    .delete(`${env.REACT_APP_URL}/device/remove/${idDevice}`)
    .then((res) => {
      const { data } = res;
      return data;
    });
}

function createDevice(deviceData) {
  return axios.post(`${env.REACT_APP_URL}/device/add`, deviceData);
}

export { deleteDevice, createDevice };
