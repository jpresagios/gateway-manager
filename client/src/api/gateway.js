import axios from "axios";
import env from "../configs/env-config";

console.log("------------------");
console.log(env.REACT_APP_URL);
console.log("------------------");

function gateWayList() {
  return axios
    .get(`${env.REACT_APP_URL}/gateway/all`)
    .then((res) => {
      const { data } = res;
      return data;
    })
    .catch((error) => {
      console.log(error);
    });
}

function createGateWay(gatewayData) {
  return axios.post(`${env.REACT_APP_URL}/gateway/insert`, gatewayData);
}

function detailGateWay(idDetail) {
  return axios
    .get(`${env.REACT_APP_URL}/gateway/detail/${idDetail}`)
    .then((res) => {
      const { data } = res;
      return data;
    });
}

export { gateWayList, createGateWay, detailGateWay };
