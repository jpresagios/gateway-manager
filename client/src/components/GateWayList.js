import React, { Component } from "react";
import { Collapse, CardBody, Card, CardHeader } from "reactstrap";
import { gateWayList } from "../api/gateway";
import DeviceList from "./Device/DeviceList";
import { Link } from "react-router-dom";

class GateWayList extends Component {
  constructor(props) {
    super(props);
    this.toggle = this.toggle.bind(this);
    this.state = { collapse: 0, gateways: [] };
  }

  componentDidMount() {
    gateWayList().then((res) => {
      const { data } = res;
      this.setState({
        gateways: data,
      });
    });
  }

  toggle(e) {
    let event = e.target.dataset.event;
    this.setState({
      collapse: this.state.collapse === Number(event) ? 0 : Number(event),
    });
  }

  render() {
    const { gateways, collapse } = this.state;

    return (
      <>
        <Link to="/gateway/add" className="btn btn-primary mb-3">
          Create GateWay
        </Link>

        {gateways.map((gateway, index) => {
          return (
            <Card style={{ marginBottom: "1rem" }} key={index}>
              <CardHeader onClick={this.toggle} data-event={index}>
                {gateway.serialNumber}

                <Link
                  to={"/gateway/detail/" + gateway._id}
                  className="float-right"
                >
                  Detail
                </Link>
              </CardHeader>
              <Collapse isOpen={collapse === index}>
                <CardBody>
                  <DeviceList devices={gateway.devices} key={gateway._id} />
                </CardBody>
              </Collapse>
            </Card>
          );
        })}
      </>
    );
  }
}

export default GateWayList;
