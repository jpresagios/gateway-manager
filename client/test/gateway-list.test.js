import GateWayList from "../src/components/GateWayList";

import React from "react";

import { render, cleanup } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
import { gateWayList } from "../src/api/gateway";
import gatewayData from "./fake-data/gateway";

jest.mock("../src/api/gateway");

gateWayList.mockResolvedValue(gatewayData);

afterEach(cleanup);

describe("GateWayList component tests", () => {
  test("Snapshot for GateWayList pass successfully", () => {
    const { asFragment } = render(
      <Router>
        <GateWayList />
      </Router>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  test("Validate API to consume Gateway List is called once", () => {
    expect(gateWayList).toHaveBeenCalledTimes(1);
  });
});
