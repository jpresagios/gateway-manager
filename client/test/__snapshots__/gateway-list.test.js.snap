// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GateWayList component tests Snapshot for GateWayList pass successfully 1`] = `
<DocumentFragment>
  <a
    class="btn btn-primary mb-3"
    href="/gateway/add"
  >
    Create GateWay
  </a>
</DocumentFragment>
`;
